virtool/samples/api.py

```diff
@@ -75,319 +90,343 @@
 routes = Routes()
 
 
-@routes.get("/samples")
-async def find(req):
-    """
-    Find samples, filtering by data passed as URL parameters.
+@routes.view("/samples")
+class SamplesView(PydanticView):
+    async def get(self) -> Union[r200[List[GetSamplesResponse]], r400]:
+        """
+        Find samples, filtering by data passed as URL parameters.
 
-    """
-    db = req.app["db"]
-    pg = req.app["pg"]
+        Status Codes:
+            200: Successful operation
+            400: Invalid query
+        """
+        db = self.request.app["db"]
+        pg = self.request.app["pg"]
+
+        v = Validator(QUERY_SCHEMA, allow_unknown=True)
 
-    v = Validator(QUERY_SCHEMA, allow_unknown=True)
+        if not v.validate(dict(self.request.query)):
+            raise InvalidQuery(v.errors)
 
-    if not v.validate(dict(req.query)):
-        raise InvalidQuery(v.errors)
+        queries = []
 
-    queries = list()
+        term = self.request.query.get("find")
 
-    term = req.query.get("find")
+        if term:
+            queries.append(compose_regex_query(term, ["name", "user.id"]))
 
-    if term:
-        queries.append(compose_regex_query(term, ["name", "user.id"]))
+        if "label" in self.request.query:
+            labels = self.request.query.getall("label")
+            labels = [int(label) if label.isdigit() else label for label in labels]
```

Comment:
Label IDs are always integers, so there is no point including non-digits in the query. We can filter the list comprehension like so:
```py
labels = [int(label) for label in labels if if label.isdigit()]
```

Anything that isn't truthy for `is_digit()` will be excluded.

https://realpython.com/lessons/filtering-elements-list-comprehensions/